py/kubeflow/testing/cleanup_ci.py

```diff
@@ -317,38 +325,38 @@ def cleanup_instance_groups(args):
       for s in results["items"]:
         name = s["name"]
         age = getAge(s["creationTimestamp"])
-
-        infra_type = name_to_infra_type(name)
-
-        if not infra_type:
-          logging.info("Skipping intance group %s; it does not match any "
-                       "infra type.", name)
+        size = s["size"]
+        if size > 0:
+          logging.info("Skipping instance group %s because it is in use by %d "
+                       "instances.", name, size)
+          in_use.append(name)
           continue
 
-        logging.info("Instance group %s categorized as %s", name, infra_type)
-
-        if age > MAX_LIFETIME[infra_type]:
-          logging.info("Deleting instanceGroups: %s, age = %r", name, age)
-          if not args.dryrun:
-            try:
-              response = instanceGroups.delete(project=args.project,
-                                               zone=zone,
-                                              instanceGroup=name).execute()
-              logging.info("response = %r", response)
-              expired.append(name)
-            except Exception as e: # pylint: disable=broad-except
-              logging.error(e)
-              in_use.append(name)
-        else:
+        infra_type = ig_name_to_infra_type(name)
+        logging.info("Instance group %s has been identified as %s", name, infra_type)
```

Comment:
It looks like you eliminated the check that infra_type is not none. Should we still be enforcing that? i.e. we only delete instances if we are able to map them to an infra type?

Reply:
done.